src/app/(proper_react)/MobileShell.tsx

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+import { ReactNode, useState } from "react";
+import Link from "next/link";
+import Image from "next/image";
+import { Session } from "next-auth";
+import styles from "./MobileShell.module.scss";
+import monitorLogo from "./images/monitor-logo.webp";
+import { CloseBigIcon, ListIcon } from "../components/server/Icons";
+import { useL10n } from "../hooks/l10n";
+import { PageLink } from "./PageLink";
+
+export type Props = {
+  children: ReactNode;
+  session: Session | null;
+};
+
+export const MobileShell = (props: Props) => {
+  const l10n = useL10n();
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  return (
+    <div
+      className={`${styles.wrapper} ${
+        isExpanded ? styles.hasOpenMenu : styles.hasClosedMenu
+      }`}
+    >
+      <header className={styles.header}>
+        <div className={styles.headerStart}>
+          <button
+            onClick={() => setIsExpanded(!isExpanded)}
+            className={styles.menuToggleButton}
+            title={l10n.getString(
+              isExpanded
+                ? "main-nav-button-collapse-tooltip"
+                : "main-nav-button-expand-tooltip"
+            )}
+          >
+            {isExpanded ? (
+              <CloseBigIcon
+                alt={l10n.getString("main-nav-button-collapse-label")}
+              />
+            ) : (
+              <ListIcon alt={l10n.getString("main-nav-button-expand-label")} />
+            )}
+          </button>
+        </div>
+        <div className={styles.headerMiddle}>
+          <Link href="/" className={styles.homeLink}>
+            <Image
+              src={monitorLogo}
+              alt={l10n.getString("main-nav-link-home-label")}
+              width={170}
+            />
+          </Link>
+        </div>
+        <div className={styles.headerEnd}>
+          {/* For the app and user menus */}
+        </div>
+      </header>
+      <div className={styles.nonHeader}>
+        <nav className={styles.mainMenuLayer}>
+          <div className={styles.mainMenu}>
```

Comment:
nit: Do we need this `<div>` or can we move the `className={styles.mainMenu}` onto the following child `<ul>` and remove a level of nesting?

Reply:
Yeah, it grated my sensibilities as well. But unfortunately, I couldn't come up with a way to do away with one. `.mainMenuLayer` is `position: sticky` to overlap the main content, but also has `height: 0` to make sure the content does not get pushed down when the menu opens. `.mainMenu` then again has a regular height that breaks out of `mainMenuLayer` but still overlaps the content.

I did not want to use the `<ul>` for `.mainMenu`, because the menu will later contain more elements than just the list of links (e.g. an "Upgrade" button).